### PR TITLE
Feature: Add ability to disable 2FA from User Management page

### PR DESCRIPTION
**Pull request by juggernot325**
## Description

This PR enables Administrator-level users to disable multi-factor authentication for other users from the user management section of the administration page. The admin will now have a new option in the actions menu to disable 2FA if the user has it enabled. After entering their password into the confirmation box and clicking confirm that user will no longer require a TOTP device to login.

Some other minor fixes that made it in:
* Standardized all user-facing text to use "Multi-Factor Authentication" or "MFA" since we were using both throughout the app. MFA is more broad and should support any additional authentication tools we add in the future.
* Disable 2FA dialogue box now clears the password field after closing or submitting
* Fixed a bug in one of our test helpers that caused a misleading error message on failure
* Some minor icon changes were made to the actions menu on the user management page
* Fix some test flakiness where client timeouts were invoking our retry logic. The retry logic had a bug where it was re-sending an empty request body, causing the test to fail due to an `EOF` error.

## Motivation and Context

* Admins can get users back into their account if their 2FA device is no longer available

## How Has This Been Tested?

* Tested manually through the UI by creating another user, logging in as that user through an Incognito browser, enabling 2FA, then disabling it from my Admin session.

## Screenshots (if appropriate):

* User actions menu updates
Screenshot 2023-08-24 at 10 38 49 AM

* Disable 2FA dialogue
Screenshot 2023-08-24 at 10 39 02 AM

* Successfully removed 2FA
Screenshot 2023-08-24 at 10 42 09 AM

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [x] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx [hunk 1]
```diff
@@ -15,19 +15,28 @@
...
+
 
     const handleOnClose = () => {
         onClose();
+        setSecret('');
+    };
+
+    const handleOnCancel = () => {
+        onCancel();
+        setSecret('');
```
Comment (urangel) [inline packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx:39]: This may be leftover from lifting the state out of this component. It would be more consistent to handle clearing the secret in the `onClose` and `onCancel` handlers passed into this component in `Users.tsx` like it was done for `onSave`.

File: cmd/ui/src/views/Users/UserActionsMenu.tsx [hunk 2]
```diff
@@ -193,7 +211,7 @@ const UserActionsMenu: React.FC<UserActionsMenuProps> = ({
                             setAnchorEl(null);
                         }}>
                         <ListItemIcon>
-                            <FontAwesomeIcon icon={faUserTimes} />
+                            <FontAwesomeIcon icon={faTrash} />
```
Comment (elikmiller) [inline cmd/ui/src/views/Users/UserActionsMenu.tsx:214]: This change seems unrelated, was it intended?

File: packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx [hunk 3]
```diff
@@ -15,19 +15,28 @@
...
 const Disable2FADialog: React.FC<{
     open: boolean;
     onClose: () => void;
     onCancel: () => void;
     onSave: (secret: string) => void;
     error?: string;
-}> = ({ open, onClose, onCancel, onSave, error }) => {
-    const [secret, setSecret] = useState('');
+    secret: string;
+    setSecret: (secret: string) => void;
```
Comment (elikmiller) [inline packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx:27]: Recommend naming this prop `onSecretChange` or similar

**Review by elikmiller — CHANGES_REQUESTED** [review]

File: packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx (hunk 3 above)
Comment (elikmiller) [reply packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx:27]: You use it in the event handler for the secret input itself which is why I think `onSecretChange` is appropriate.

If you want to clear the value of the secret when the dialog closes or the form is submitted you can do that by setting that value in the onClose, onCancel, onSave callbacks that get passed in as props.

Comment (elikmiller) [reply packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx:27]: To that end I think the handleOnClose and handleOnCancel functions below are not really necessary

Comment (juggernot325) [reply packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx:27]: Ahh, I see what you mean now. Line 59 of that file actually existed prior to my changes. I'm not sure why it was set up this way as I'm still a noob with React. I do think that the `setSecret` intent has become conflated due to my changes, though.

**Review by maffkipp — APPROVED** [review]

**Review by elikmiller — APPROVED** [review]
